File: ScaleformUI_Lua/src/Menus/PauseMenus/PauseMenu/Items/GalleryItem.lua
GalleryItem = {}
GalleryItem.__index = GalleryItem
setmetatable(GalleryItem, { __index = PauseMenuItem })
GalleryItem.__call = function() return "GalleryItem" end


---@class GalleryItem
---@field public TextureDictionary string
---@field public TextureName string
---@field public Label1 string
---@field public Label2 string
---@field public Label3 string
---@field public Label4 string
---@field public RightPanelDescription string
---@field public Parent GalleryTab
---@field public Blip FakeBlip
function GalleryItem.New(txd, txn, label1, label2, label3, label4)
    local base = PauseMenuItem.New("")
    base.TextureDictionary = txd or ""
    base.TextureName = txn or ""
    base.Label1 = label1 or ""
    base.Label2 = label2 or ""
    base.Label3 = label3 or ""
    base.Label4 = label4 or ""
    base.RightPanelDescription = ""
    base.Blip = nil
    base.Activated = function(tab, item, totalIndex, gridIndex)
    end
    return setmetatable(base, GalleryItem)
end

---Sets item description labels.
---@param label1 string
---@param label2 SColor
---@param label3 boolean
---@param label4 boolean
function GalleryItem:SetLabels(label1, label2, label3, label4)
    self.Label1 = label1
    self.Label2 = label2
    self.Label3 = label3
    self.Label4 = label4
    if self.Parent ~= nil and self.Parent.Base ~= nil and self.Parent.Base.Parent ~= nil and self.Parent.Base.Parent:Visible() and self.Parent.Visible then
        if self.Parent:isItemVisible(IndexOf(self.Parent.GalleryItems, self)) then
            local gridPosition = self.Parent:gridIndexFromItemIndex(IndexOf(self.Parent.GalleryItems, self))
            ScaleformUI.Scaleforms._pauseMenu._pause:CallFunction("UPDATE_GALLERY_ITEM", gridPosition - 1, gridPosition - 1, 33, 4, 0, 1, self.Label1, self.Label2, self.TextureDictionary, self.TextureName, 1, false, self.Label3, self.Label4)
        end
    end
end

---Sets item long panel description.
---@param description string
function GalleryItem:SetRightDescription(description)
    self.RightPanelDescription = description
    if self.Blip ~= nil then return end
    if self.Parent ~= nil and self.Parent.Base ~= nil and self.Parent.Base.Parent ~= nil and self.Parent.Base.Parent:Visible() and self.Parent.Visible then
        if self.Parent:isItemVisible(IndexOf(self.Parent.GalleryItems, self)) then
            AddTextEntry("gallerytab_desc", self.RightPanelDescription)
            ScaleformUI.Scaleforms._pauseMenu._pause:CallFunction("SET_GALLERY_PANEL_HIDDEN", false)
            BeginScaleformMovieMethod(ScaleformUI.Scaleforms._pauseMenu._pause.handle, "SET_GALLERY_PANEL_DESCRIPTION")
            BeginTextCommandScaleformString("gallerytab_desc")
            EndTextCommandScaleformString_2()
            EndScaleformMovieMethod()
        end
    end
end

File: ScaleformUI_Csharp/Menus/Pause Menus/Elements/Items/GalleryItem.cs
using ScaleformUI.Elements;
using ScaleformUI.PauseMenu;
using static CitizenFX.Core.Native.API;

namespace ScaleformUI.PauseMenus.Elements.Items
{
    public class GalleryItem : PauseMenuItem
    {
        public string TextureDictionary { get; private set; }
        public string TextureName { get; private set; }
        public string Label1 { get; private set; }
        public string Label2 { get; private set; }
        public string Label3 { get; private set; }
        public string Label4 { get; private set; }
        public ScaleformLabel RightPanelDescription { get; private set; }

        public event GalleryItemSelected Activated;

        public FakeBlip Blip { get; set; }

        public GalleryItem(string textureDictionary, string textureName) : base("")
        {
            TextureDictionary = textureDictionary;
            TextureName = textureName;
        }

        public void SetLabels(string label1, string label2, string label3, string label4)
        {
            Label1 = label1;
            Label2 = label2;
            Label3 = label3;
            Label4 = label4;
            //if (Parent != null && Parent.)
            //{
            //    int gridPosition = Parent.GridIndexFromItemIndex(Parent.GalleryItems.IndexOf(this));
            //    Parent.Parent._pause._pause.CallFunction("UPDATE_GALLERY_ITEM", gridPosition, gridPosition, 33, 4, 0, 1, Label1, Label2, TextureDictionary, TextureName, 1, false, Label3, Label4);
            //}
        }
        public void SetRightDescription(string description)
        {
            RightPanelDescription = description;
            if (Blip != null) return;
            //if (Parent != null && Parent.Visible && Parent.IsItemVisible(Parent.GalleryItems.IndexOf(this)))
            //{
            //    AddTextEntry("gallerytab_desc", RightPanelDescription);
            //    Parent.Parent._pause._pause.CallFunction("SET_GALLERY_PANEL_HIDDEN", false);
            //    BeginScaleformMovieMethod(Parent.Parent._pause._pause.Handle, "SET_GALLERY_PANEL_DESCRIPTION");
            //    BeginTextCommandScaleformString("gallerytab_desc");
            //    EndTextCommandScaleformString_2();
            //    EndScaleformMovieMethod();
            //}
        }

        internal void ItemSelected(GalleryTab tab, GalleryItem item, int totalIndex, int gridIndex)
        {
            Activated?.Invoke(tab, item, totalIndex, gridIndex);
        }
    }
}
